src/i18n.ts

```diff
@@ -37,17 +14,11 @@ export async function getTranslations(localeCode: string) {
     });
 }
 
-const dev = import.meta.env.DEV;
-
 export function initi18n(resources: Resource, locale: string) {
-  const languageDetector = new LanguageDetector();
-  languageDetector.addDetector(detector);
-
-  const lookupOrder = dev ? ["querystring"] : ["pagesJSDetector"];
+  const lookupOrder = ["querystring"];
```

Comment:
Is this okay to have it just be querystring? I'd expect that to work when serving locally because ?locale= is required to change the locale but not on prod. Perhaps the code that used to take the locale from __INITIAL__DATA__.document.locale could instead grab the locale from the lang attribute of the root html element?

Reply:
Honestly I think I should remove this part as well as it looks like the lookupOrder is only used in the detection property of the i18n.init options. The lang detection seems to be working properly atm on the staging site [here](https://analytics-v2--all-accor--directory-com-pagescdn-com.preview.pagescdn.com/fr/monde) b.c. I think as long as we are passing in the locale to the i18n.changeLanguage(locale) setup it should be gtg.

Just double checked and I think the options.detection params are only used by the i18n-language-detector package (see defaults set in their repo [here](https://github.com/i18next/i18next-browser-languageDetector/blob/9efebe6ca0271c3797bc09b84babf1ba2d9b4dbb/src/index.js#L11))

^^^same comment as accor specific PR lol

Reply:
I agree if we're able to remove the __INITIAL__DATA__ version of the detector then we should also be able to remove the query param one as well and skip all of it. 

I'm not confident enough in how the i18n library works in general though to say just from looking at the code change. 

I would try removing it and test each of local dev and a deployed site with javascript enabled / disabled and make sure that in each of the 4 cases we're getting the correct language. It sounds like you may have already verified on Accor though?

Reply:
yeah ill remove the rest of the language detector for now.  Its working on accor atm but I havent tested it w/javascript disabled I can do that now

Reply:
Thanks for doing the testing!